File: examples/torch/hello-torch/hello-torch.py
import torch

a = torch.ones(3)
print(a)

a[2] = 2.0
print(a)

points = torch.tensor([4.0, 1.0, 5.0, 3.0, 2.0, 1.0])
print(points)

points = torch.tensor([[4.0, 1.0], [5.0, 3.0], [2.0, 1.0]])
print(points)

print(points[0])

points_storage = points.storage()
print(points_storage)
print(points_storage[0])

second_point = points[1]
print(second_point.storage_offset())
print(second_point.stride())

print(points.t())

points_gpu = points.to(device='cuda')
print(points_gpu)

File: examples/torch/hello-torch/hello-torch.cpp
#include <iostream>
#include <torch/torch.h>

using namespace torch;
using namespace std;

int main(int argc, char** argv)
{
    Tensor a = torch::ones(3);
    cout << a << endl;
    a[2] = 2.0;
    cout << a << endl;

    Tensor points = torch::tensor({4, 1, 5, 3, 2, 1});
    cout << points << endl;

    points = torch::tensor({{4, 1}, {5, 3}, {2, 1}});
    cout << points << endl;
    cout << points[0] << endl;
    cout << points.storage() << endl;

    Tensor second_points = points[1];
    cout << "storage_offset: " << second_points.storage_offset() << endl;
    cout << "stride: " << second_points.stride(0) << endl;
    cout << points.t() << endl;

    cout << "CUDA available: " << torch::cuda::is_available() << endl;

    cout << "requires grad: " << points.requires_grad() << endl;
    cout << "is inference: " << points.is_inference() << endl;

    return 0;
}